backend/models/notification.js

```diff
@@ -401,7 +401,7 @@ module.exports = {
 	insertUserReferencedNotification: async function (referrer, teamspace, modelId, type, _id, referee) {
 		try {
 			const user = await User.findByUserName(referee);
-			if (await user.isMemberOfTeamspace(teamspace)) {
+			if (User.isMemberOfTeamspace(user, teamspace)) {
```

Comment:
similar sentiment here, notification.js shouldn't need to get the whole user data out just to do `isMemberOfTeamspace`.

`isMemberOfTeamspace` can fetch that data out with projections to save IO and also less coupling

Reply:
As above isMemberOfTeamspace is taking a user in order to reuse the fetch instead of fetching the user multiple times. In this case user is only being use once. I could add isMemberOfTeamspaceByUsername.

Reply:
Once again, here the code is only fetching User just to call that function.

Considering User consist of a lot of data (including an image for avatar), I think it's a lot smarter to let isMemberOfTeamspace fetch it with a projection.

I agree if it has to reuse it somewhere, but so far none of it is the case. So I think we don't need the original function, just replace it with the one that takes it as a string